### PR TITLE
adds upgrade wand, magicman balance changes

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
adds the upgrade wand, a 5 tc traitor item for magicians. you can use it on any magician item to upgrade its stats. 
lowers teleglove range by 1
changes tophat calcs a bit, it's a bit faster on non-carbons now, and is very fast when you use it on yourself
## Why It's Good For The Game
fun magicman shit

## Changelog
:cl:
add: the upgrade wand, a magician only traitor item that upgrades his gear
balance: lowers teleglove range by 1
balance: bottomless top hat now kidnaps non-carbons faster, and works better on the user
/:cl:

### CODE REVIEW

File: code/game/objects/items/miscellaneous.dm [hunk 1]
```diff
@@ -180,13 +180,30 @@
...
+		maximum_size = 4
+		to_chat(user, "<span_class='notice'>You upgrade the [src] with the [wand]</span>")
+		playsound(user, 'sound/weapons/emitter2.ogg', 25, 1, -1)
+	
 /obj/item/clothing/head/that/bluespace/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
 	if(isliving(target))
 		var/mob/living/M = target
+		var/kidnaptime = max(10, (M.health * (M.mob_size / 2)))
+		if(iscarbon(target))
```
Comment (PowerfulBacon) [inline code/game/objects/items/miscellaneous.dm:202]: Should silicons have an extra time too, if carbons get it?

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 2]
```diff
@@ -88,7 +88,16 @@
 	icon_state = "white"
 	item_state = "wgloves"
 	item_color="white"
-	var/range = 4
+	var/range = 3
+
+/obj/item/clothing/gloves/color/white/magic/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))
```
Comment (PowerfulBacon) [inline code/modules/clothing/gloves/miscellaneous.dm:95]: Check to see if already upgraded.

File: code/game/objects/items/miscellaneous.dm [hunk 3]
```diff
@@ -180,13 +180,30 @@
 	var/itemheld = FALSE
 	var/capacity = 2
 	var/maximum_size = 2 //one human, two pets, unlimited tiny mobs, but no big boys like megafauna
+	var/kidnappingcoefficient = 1
 
+/obj/item/clothing/head/that/bluespace/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))
```
Comment (PowerfulBacon) [inline code/game/objects/items/miscellaneous.dm:187]: You should add a check to see if it is already upgraded before people buy 2 wands and try to double upgrade things.

File: code/game/objects/items/miscellaneous.dm [hunk 4]
```diff
@@ -180,13 +180,30 @@
...
+		var/obj/item/upgradewand/wand = W
+		wand.used = TRUE
+		kidnappingcoefficient = 0.5
+		capacity = 4
+		maximum_size = 4
+		to_chat(user, "<span_class='notice'>You upgrade the [src] with the [wand]</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/miscellaneous.dm]: Add punctuation to the end of this sentence.

File: code/game/objects/items/devices/traitordevices.dm [hunk 5]
```diff
@@ -255,6 +255,16 @@ effective or pretty fucking useless.
...
 	attack_verb = null
 
+/obj/item/shadowcloak/magician/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))
+		var/obj/item/upgradewand/wand = W
+		wand.used = TRUE
+		charge = 450
+		max_charge = 450
+		to_chat(user, "<span_class='notice'>You upgrade the [src] with the [wand]</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/traitordevices.dm]: Add punctuation to the end of this sentence.

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 6]
```diff
@@ -88,7 +88,16 @@
...
-	var/range = 4
+	var/range = 3
+
+/obj/item/clothing/gloves/color/white/magic/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))
+		var/obj/item/upgradewand/wand = W
+		wand.used = TRUE
+		range = 6
+		to_chat(user, "<span_class='notice'>You upgrade the [src] with the [wand]</span>")
```
Comment (PowerfulBacon) [inline code/modules/clothing/gloves/miscellaneous.dm]: Add punctuation to the end of this sentence.

File: code/game/objects/items/devices/traitordevices.dm [hunk 7]
```diff
@@ -255,6 +255,16 @@ effective or pretty fucking useless.
 	equipslot = SLOT_NECK
 	attack_verb = null
 
+/obj/item/shadowcloak/magician/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/upgradewand))
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/traitordevices.dm:260]: You should add a check to see if it is already upgraded before people buy 2 wands and try to double upgrade things, to a message saying they updated it with no actual effect.

File: code/game/objects/items/miscellaneous.dm [hunk 8]
```diff
@@ -277,3 +294,12 @@
...
 	return ouija_spaghetti_list
+
+/obj/item/upgradewand
+	desc = "A wand laced with nanotech calibration devices."
+	name = "Upgrade Wand"
+	icon = 'icons/obj/guns/magic.dmi'
+	icon_state = "nothingwand"
+	item_state = "wand"
+	w_class = WEIGHT_CLASS_SMALL
+	var/used = FALSE
```
Comment (PowerfulBacon) [inline code/game/objects/items/miscellaneous.dm:306]: This variable is never actually used, only set. Once upgraded, it gets set to true, but there is no check that requires this to be false to upgrade something

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/miscellaneous.dm (hunk 1 above)
Comment (zeskorion) [reply code/game/objects/items/miscellaneous.dm:202]: Carbon's extra time is due to how being in crit counts as negative HP

Comment (PowerfulBacon) [reply code/game/objects/items/miscellaneous.dm:202]: Makes sense

File: code/game/objects/items/miscellaneous.dm [hunk 9]
```diff
@@ -277,3 +295,12 @@
 			var/atom/A = V
 			ouija_spaghetti_list[initial(A.name)] = A
 	return ouija_spaghetti_list
+
+/obj/item/upgradewand
+	desc = "A wand laced with nanotech calibration devices."
```
Comment (ike709) [inline code/game/objects/items/miscellaneous.dm]: This needs to specify that it should be used on magician gear.

**Pull request merged**